dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/HdfsPermissionExtractionTask.java

```diff
@@ -68,7 +69,7 @@ public Void run(TaskRunContext context) throws Exception {
     return null;
   }
 
-  private void doRun(Writer output) throws IOException {
+  private void doRun(Writer output) throws IOException, ExecutionException, InterruptedException {
```

Comment:
Thanks for the lesson;
(I did not know about the existence of @WillClose)

Reply:
Thank you, this is good. I suspect that relatively few of the static analyzers honour it, but even if we consider it to be purely documentation, I think annotations are a nice, formally-analyzable-on-demand way of writing that documentation, so I tend to write them rather than javadoc. So I don't think the existence of a specific static analyzer should be a prerequisite for writing the annotation. </opinion>